crates/solver/src/settlement_rater.rs

```diff
@@ -140,7 +149,34 @@ impl SettlementRating for SettlementRater {
         prices: &ExternalPrices,
         gas_price: GasPrice1559,
     ) -> Result<(Vec<RatedSolverSettlement>, Vec<SimulationWithError>)> {
-        let simulations = self.simulate_settlements(settlements, gas_price).await?;
+        // first simulate settlements without internalizations to make sure they pass
+        let simulations = self
+            .simulate_settlements(
+                settlements,
+                gas_price,
+                InternalizationStrategy::EncodeAllInteractions,
```

Comment:
This means we should do the competition based on the simulation without internalizing interactions, let me fix this.

Reply:
So, I reverted the logic:
- Fist we do simulation with internalization to make sure those settlements can be settled onchain.
- For those that can be settled onchain, rerun the simulation without internalization for solver competition.

Reply:
Tagging @harisang to be aware of this change since I think it is not how it was intended to be done in prioritization document where it was explained as: 

> First, generate a solution with NO internalizations; this would require an additional change in the json response of the solvers, since they would now have to provide an execution plan for all interactions. The goal is to simulate the solution without the internalization in order to verify it goes through.
> If step (a) succeeds, only then the driver will generate the calldata with the internalizations, and then rank the solution as it does now.

Reply:
> Don't know if this is already taken into account when building access lists or not, but I do think we need to make sure both simulations succeed.

Access lists are dependent on internalization strategy and are estimated separately for each type of simulations so no issue with that.

Reply:
Reverted back to original solution.